src/api/process.c

```diff
@@ -65,7 +65,7 @@ typedef enum {
 #endif
 
 static bool poll_process(process_t* proc, int timeout) {
-  if (!proc->running)
+  if (!proc->running || timeout < 0)
```

Comment:
Oh, also, negative timeouts are defines here, though. So we shouldn't have this < 0 statement.

Reply:
Was not aware of this because I had not looked too deeply, but now that you mention it this is correct.

I swapped the logic to ensure the ticks are all signed instead